src/movies.js

```diff
@@ -1,28 +1,105 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors (array){
+  const directors = array.map((movie) => {
+    return movie.director
+  })
+  return directors
+}
+function filterDirectors(item){
+  const directors= getAllDirectors(item);
+  const filteredDirectors=[];
+  directors.forEach((director)=>{
+    if( !filteredDirectors.includes(director)){
+      filteredDirectors.push(director)
+    }
+  })
+  return filteredDirectors
+}
 
-// Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
 
+// Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+function howManyMovies(countMovies) {
+  const dramaMoviesByStevenSpilberg=countMovies.filter((movie) =>{
+    return movie.director ==="Steven Spielberg" && movie.genre.includes("Drama");
+  });
+  return dramaMoviesByStevenSpilberg.length;
+}
+  
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(movies) {
+  if(movies.length===0){
+    return 0;
+  }
+const sumScores=movies.reduce((accumulator, movie)=>{
+if(movie.score){
+return accumulator + movie.score;
+}else {
+  return accumulator;
+}
+}, 0);
+averageScore = sumScores / movies.length;
+return +(Math.round(averageScore + "e+2")  + "e-2");
```

Comment:
Parece que funciona, pero tal vez es más facil de entender si ponemos:
```suggestion
return Number(averageScore.toFixed(2));
```